code/modules/mob/living/carbon/brain/MMI.dm

```diff
@@ -90,25 +90,28 @@
 		return
 
 	// Maybe later add encryption key support, but that's a pain in the neck atm
-	if(isscrewdriver(O))
-		if(radio)
-			user.visible_message("<span class='warning'>[user] begins to uninstall the radio from [src]...</span>", \
-								 "<span class='notice'>You start to uninstall the radio from [src]...</span>")
-			if(do_after(user, 40 * O.toolspeed, target = src))
-				uninstall_radio()
-				new /obj/item/mmi_radio_upgrade(get_turf(src))
-				user.visible_message("<span class='warning'>[user] uninstalls the radio from [src].</span>", \
-									 "<span class='notice'>You uninstall the radio from [src].</span>")
-		else
-			to_chat(user, "<span class='warning'>There is no radio in [src]!</span>")
-		return
 
 	if(brainmob)
 		O.attack(brainmob, user)//Oh noooeeeee
 		// Brainmobs can take damage, but they can't actually die. Maybe should fix.
 		return
-	..()
+	return ..()
 
+/obj/item/mmi/screwdriver_act(mob/user, obj/item/I)
+	. = TRUE
+	if(!I.tool_use_check(user, 0))
+		return
+	if(!radio)
+		to_chat(user, "<span class='warning'>There is no radio in [src]!</span>")
+		return
+	user.visible_message("<span class='warning'>[user] begins to uninstall the radio from [src]...</span>", \
+							 "<span class='notice'>You start to uninstall the radio from [src]...</span>")
+	if(!I.use_tool(src, user, 40, volume = I.tool_volume) || !radio)

```

Comment:
ah, good catch on the radio check here

Reply:
I think it already checked if there was a radio , it's just buried in that massive proc 😄